### DISCUSSION TITLE
Example of chain aware provider?

### DISCUSSION THREAD

**Discussion by JuanSanchez420** [wevm/wagmi#1793] [Ideas]
Could we get a bare bones example of how to set up a chain aware provider? I'm not only confused on how to set this up, I'm confused on how this wasn't built into WAGMI since it's a common design pattern: fetching data on load without a wallet connected. Ethers.js even distinguishes between providers and signers, and you just left out one of them?

**Comment by jxom** [comment]
The most simplest way to set up a chain aware provider would be to follow the Getting Started guide on the docs. The Configure Chains step shows you how to set up wagmi’s built-in chain aware provider: https://wagmi.sh/react/getting-started#configure-chains

**Comment by JuanSanchez420** [comment]
The documentation isn't helpful. If the documentation was clear, you wouldn't keep getting this question. I've gone through it as well as all the closed tickets and discussions. I finally figured it out:

![image](https://user-images.githubusercontent.com/101058183/216794363-c9b17d26-fdb4-4ee8-aafa-d33b143ba5fa.png)

Also remember: pass chainId with every function/hook you use otherwise you'll get null values back.

![image](https://user-images.githubusercontent.com/101058183/216794444-0ae199d2-69f8-4eeb-8dd9-f72e75419524.png)
